### PR TITLE
Remove Game.ModData references from Widget code.

### PR DESCRIPTION
**Pull request by pchote**
The main focus of this change is code hygiene (relying on a static reference for something which may change isn't nice), but has a side-effect of adding support for map-defined resources in the editor.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs [hunk 1]
```diff
@@ -67,11 +68,12 @@ public string ProgressLabelText()
...
 
 		[ObjectCreator.UseCtor]
-		public MultiplayerLogic(Widget widget, Action onStart, Action onExit, string directConnectHost, int directConnectPort)
+		public MultiplayerLogic(Widget widget, ModData modData, Action onStart, Action onExit, string directConnectHost, int directConnectPort)
 		{
+			this.modData = modData;
 			this.onStart = onStart;
 			this.onExit = onExit;
-
+
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs]: Lonely tab.

**Comment by pchote** [comment]
Fixed.

**Comment by Mailaender** [comment]
:+1:

**Comment by obrakmann** [comment]
```
OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs:L28: [SA1215] All non-static readonly private fields must be placed before all non-static non-readonly private fields.

OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs:L29: [SA1215] All non-static readonly private fields must be placed before all non-static non-readonly private fields.
```

**Comment by pchote** [comment]
Fixed.

**Comment by obrakmann** [comment]
Needs a rebase as well

**Comment by pchote** [comment]
Rebased.

**Comment by obrakmann** [comment]
Looks good to me, :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/6292ca8cafbd6e63705c174285d535232d6f2ab6)

**Pull request merged**